fe/fe-core/src/main/java/com/starrocks/qe/recursivecte/RecursiveCTEExecutor.java

```diff
@@ -0,0 +1,384 @@
+            sb.append("Temporary Table: ").append(group.tempTableStmt.getTableName()).append("\n");
+            sb.append("Start Statement: ").append(AstToSQLBuilder.toSQL(group.startStmt)).append("\n");
+            sb.append("Recursive Statement: ").append(AstToSQLBuilder.toSQL(group.recursiveStmt)).append("\n");
+            sb.append("--------------------------------------------------\n");
+        }
+
+        sb.append("Outer Statement After Rewriting:\n");
+        sb.append(AstToSQLBuilder.toSQL(stmt)).append("\n");
+        return sb.toString();
+    }
+
+
+    private void analyze(StatementBase stmt, ConnectContext context) {
+        try (PlannerMetaLocker locker = new PlannerMetaLocker(context, stmt)) {
+            locker.lock();
+            Analyzer.analyze(stmt, context);
+        }
+    }
+
+    public void prepareRecursiveCTE() throws Exception {
+        UUID origUUID = connectContext.getQueryId();
+        try {
+            prepareRecursiveCTEImpl();
+        } catch (Exception e) {
+            LOG.warn("Error occurred during preparing recursive CTE", e);
+            throw e;
+        } finally {
+            connectContext.setQueryId(origUUID);
+            connectContext.setExecutionId(UUIDUtil.toTUniqueId(origUUID));
+        }
+    }
+
+    private void prepareRecursiveCTEImpl() throws Exception {
+        StmtExecutor executor;
+        for (String cteName : recursiveCTEGroups.keySet()) {
+            RecursiveCTEGroup group = recursiveCTEGroups.get(cteName);
+            TableName tempTableName = cteTempTableMap.get(cteName);
+            // temp table creation
+            analyze(group.tempTableStmt, connectContext);
+            connectContext.setQueryId(UUIDUtil.genUUID());
+            connectContext.setExecutionId(UUIDUtil.toTUniqueId(connectContext.getQueryId()));
+            DDLStmtExecutor.execute(group.tempTableStmt, connectContext);
+
+            // start plan
+            InsertStmt insert = new InsertStmt(tempTableName, group.startStmt);
+            insert.setOrigStmt(new OriginStatement(AstToSQLBuilder.toSQL(insert)));
+            executor = StmtExecutor.newInternalExecutor(connectContext, insert);
+            connectContext.setQueryId(UUIDUtil.genUUID());
+            executor.execute();
+            if (!connectContext.getState().getErrType().equals(QueryState.ErrType.UNKNOWN)) {
+                LOG.warn("Error occurred during executing recursive CTE start statement: {}",
+                        connectContext.getState().getErrorMessage());
+                throw new RuntimeException("Error occurred during executing recursive CTE start statement: " +
+                        connectContext.getState().getErrorMessage());
+            }
+
+            // recursive plan
+            RecursiveCTEStager stager = new RecursiveCTEStager(group, tempTableName);
+            for (int i = 1; i < connectContext.getSessionVariable().getRecursiveCteMaxDepth(); i++) {
```

Comment:
If the loop exits due to reaching `max_depth` , the user gets incomplete results without any indication.

Reply:
okkk，next pr resolve